src/components/partials/RichModDisplay.tsx

```diff
@@ -41,50 +17,72 @@ interface Props {
 }
 
 export const RichModDisplay = ({ data, buttonType, onClick }: Props) => {
-  if (onClick) {
-    if (buttonType === 'delete') {
-      return (
-        <div className="group flex space-x-4 rounded-sm bg-transparent p-4 text-left transition-colors hover:bg-zinc-50 dark:hover:bg-zinc-800">
-          <SomeDetails data={data} />
-          <button
-            className="flex-grow-0 self-center rounded-md p-2"
-            type="button"
-          >
-            <TrashIcon
-              className="block h-5 w-5 text-red-500"
-              onClick={onClick}
+  return (
+    <div className="flex justify-between rounded-2xl border-none bg-zinc-100 p-4 pt-4 shadow-sm dark:bg-zinc-800">
+      <div className="flex w-full">
+        <div className="shrink-0">
+          {data.iconUrl && (
+            <img
+              src={data.iconUrl}
+              alt={`Icon of ${data.name}`}
+              className="mr-4 h-16 w-16 rounded-2xl object-contain"
             />
-          </button>
+          )}
         </div>
-      );
-    } else if (buttonType === 'add') {
-      return (
-        <div className="group flex space-x-4 rounded-sm bg-transparent p-4 text-left transition-colors hover:bg-zinc-50 dark:hover:bg-zinc-800">
-          <SomeDetails data={data} />
-          <button
-            className="flex-grow-0 self-center rounded-md p-2"
-            type="button"
-          >
-            <PlusIcon
-              className="block h-5 w-5 text-green-500"
-              onClick={onClick}
-            />
-          </button>
+        <div className="flex grow justify-between">
+          <div className="flex flex-col">
+            <div className="flex shrink-0 flex-wrap align-baseline">
+              <h2 className="mr-2 text-xl font-bold">{data.name}</h2>
+            </div>
+            <p className="my-0.5 font-medium">{data.description}</p>
+            <div className="flex flex-wrap">
+              <a
+                className="flex items-center underline"
+                href={data.href}
+                target="_blank"
+                rel="noreferrer noopener"
+              >
+                {providerFormat(data.provider)}
+                <ArrowTopRightOnSquareIcon className="mr-auto ml-1 h-4 w-4" />
+              </a>
+            </div>
+          </div>
+          <div className="min-w-fit text-right">
```

Comment:
on mobile this is still rendered right to the description, resulting in it being a bit squished; could you move it to be displayed under the description (like Modrinth does) on mobile?